### PR TITLE
Allow users to change their passwords.

### LINKED ISSUE #189: add Reset Password to Edit Profile
part of epics #3, #105, #4

**What type of user is this for? [volunteer/supervisor/admin/all]**
all (but most pressingly, volunteers)

**Where does/should this occur?**
On the <a href="https://casa-r4g-staging.herokuapp.com/users/edit">edit profile</a> page for all users.

**Description**
Add "Reset password" field to Edit Profile page.

**Related issues**
#125

### PR DESCRIPTION
**Pull request by seanmarcia**
Add in some error helpers. Font Awesome. And a bit more.

This resolves #189 

At the core this resolves #189 allowing folks to change their password. We already have the ability for folks to reset their password if they forgot it so the work done, primarily, is for logged in folks.

A summary of stuff with pictures:
-Changed the forgot password button link on the sign-in page to a button to make it more obvious it exists:

image

becomes: 

image

Next I changed the edit profile page around a bit:

edit_profit

becomes:

new_profile

When you click the change password button it opens up the following:

change_password

I also added in a helper method with styling for the error messages so they are not plain text and more visible. 
So, rather than doing something like this in our views:
```
      <% if @user.errors.any? %>	
        <div id="error_explanation">	
          <h2><%= pluralize(user.errors.count, "error") %> prohibited this user from being saved:</h2>	

          <ul>	
            <% user.errors.full_messages.each do |message| %>	
              <li><%= message %></li>	
            <% end %>	
          </ul>	
        </div>	
      <% end %>
```
We can just do `<%= errors_for(@user) %>`

Error messages now look like this: 

error messages

I also updated the flash messages how they are shown in the `application.html.erb` and made them dismissible. I also added in `font-awesome-rails` to the gemfile to put in an X into the alerts. (Plus it is handy to use elsewhere)

I also put in a tiny javascript function that will auto close alerts from pages after a couple of seconds.

The new alerts look like this:

alerts

It is still a WIP because I have a couple questions:

How is the team testing stuff that overlaps with gems? I'm happy to add in tests for the changing password functionality but it kind of feels like I'm testing stuff that overlaps with devise and didn't want to add cruft to the app.

I also added in Font Awesome which you all may not be cool with.

I'm also not sure where that tiny js function should live -- it is currently in `application.js`.

Let me know what you want me to do with all this and I'll move it forward.

### CODE REVIEW

**Comment by compwron** [comment]
@seanmarcia This is great! 💖
> How is the team testing stuff that overlaps with gems? I'm happy to add in tests for the changing password functionality but it kind of feels like I'm testing stuff that overlaps with devise and didn't want to add cruft to the app.
There's no particular approach right now. I agree that Devise doesn't need testing. It would be nice to have a test around "go to page X and then Devise does Y" so probably a capybara test would be best. 

> I also added in Font Awesome which you all may not be cool with.
I like it!

> I'm also not sure where that tiny js function should live -- it is currently in application.js.
That's an ok place for now. We could add a file in `js/src`

Approved! ✨🙌🏼💖

File: app/assets/stylesheets/shared/flashes.scss [hunk 1]
```diff
@@ -3,3 +3,7 @@
   padding: 8px;
   margin: 12px 0;
 }
+
+.alert-dismissible .close {
+  padding: 0.5rem 0.75rem;
+}
```
Comment (compwron) [inline app/assets/stylesheets/shared/flashes.scss:9]: nit: missing linebreak

Comment (seanmarcia) [reply app/assets/stylesheets/shared/flashes.scss]: Fixed :)

**Comment by seanmarcia** [comment]
Also added in a couple capybara system specs.

**Pull request merged**